Missing <cstdlib> include in vulkan.hpp for std::getenv causes build failures with C++ modules

vulkan.hpp makes a call to std::getenv when attempting to resolve the Vulkan library path on macOS, but it does not include <cstdlib>. 

It is transitively/implicitly included from other headers so will be fine for non-module compiles, which is why it was working before. But when using strict C++ modules it causes a compile error, because visibility and unit boundaries are more tightly enforced.

Specifically this is being seen during a dependency roll for Chromium (on MacOS bots), causing errors like this:
```
   1 ../../third_party/swiftshader/include/vulkan/vulkan.hpp:22912:37: error: declaration of 'getenv' must be imported from module
     '//build/modules:system.std.cstdlib' before it is required
   2  22912 |           if ( !m_library && ( std::getenv( "DYLD_FALLBACK_LIBRARY_PATH" ) == NULL ) )
   3        |                                     ^
   4 ../../third_party/libc++/src/include/cstdlib:130:9: note: declaration here is not visible
   5   130 | using ::getenv _LIBCPP_USING_IF_EXISTS;
```

I believe that this should be resolvable by making sure that `#include <cstdlib>` is added to the system library includes at the top of vulkan.hpp

--- a/generator/snippets/DynamicLoader.hpp
+++ b/generator/snippets/DynamicLoader.hpp
@@ -31,32 +31,32 @@
         {
           m_library = dlopen( "libvulkan.1.dylib", RTLD_NOW | RTLD_LOCAL );
         }
-		// modern versions of macOS don't search /usr/local/lib automatically contrary to what man dlopen says
-		// Vulkan SDK uses this as the system-wide installation location, so we're going to fallback to this if all else fails
-		if ( !m_library && ( std::getenv("DYLD_FALLBACK_LIBRARY_PATH") == NULL ) )
-		{
-		  m_library = dlopen( "/usr/local/lib/libvulkan.dylib", RTLD_NOW | RTLD_LOCAL );
-		}
+        // modern versions of macOS don't search /usr/local/lib automatically contrary to what man dlopen says
+        // Vulkan SDK uses this as the system-wide installation location, so we're going to fallback to this if all else fails
+        if ( !m_library && ( std::getenv( "DYLD_FALLBACK_LIBRARY_PATH" ) == NULL ) )
+        {
+          m_library = dlopen( "/usr/local/lib/libvulkan.dylib", RTLD_NOW | RTLD_LOCAL );
+        }
         // for Homebrew installations on Apple Silicon
         if ( !m_library )
-		{
-		  m_library = dlopen( "/opt/homebrew/lib/libvulkan.dylib", RTLD_NOW | RTLD_LOCAL );
-		}
+        {
+          m_library = dlopen( "/opt/homebrew/lib/libvulkan.dylib", RTLD_NOW | RTLD_LOCAL );
+        }
         if ( !m_library )
-		{
-		  m_library = dlopen( "libMoltenVK.dylib", RTLD_NOW | RTLD_LOCAL );
-		}
-		// Add support for using Vulkan and MoltenVK in a Framework. App store rules for iOS
-		// strictly enforce no .dylib's. If they aren't found it just falls through
-		if ( !m_library )
-		{
-		  m_library = dlopen( "vulkan.framework/vulkan", RTLD_NOW | RTLD_LOCAL );
-		}
-		if ( !m_library )
-		{
-		  m_library = dlopen( "MoltenVK.framework/MoltenVK", RTLD_NOW | RTLD_LOCAL );
-		}
-#  elif defined( __unix__ ) || defined( __QNX__ ) || defined(__Fuchsia__)
+        {
+          m_library = dlopen( "libMoltenVK.dylib", RTLD_NOW | RTLD_LOCAL );
+        }
+        // Add support for using Vulkan and MoltenVK in a Framework. App store rules for iOS
+        // strictly enforce no .dylib's. If they aren't found it just falls through
+        if ( !m_library )
+        {
+          m_library = dlopen( "vulkan.framework/vulkan", RTLD_NOW | RTLD_LOCAL );
+        }
+        if ( !m_library )
+        {
+          m_library = dlopen( "MoltenVK.framework/MoltenVK", RTLD_NOW | RTLD_LOCAL );
+        }
+#  elif defined( __unix__ ) || defined( __QNX__ ) || defined( __Fuchsia__ )
         m_library = dlopen( "libvulkan.so", RTLD_NOW | RTLD_LOCAL );
         if ( !m_library )
         {

--- a/generator/snippets/includes.hpp
+++ b/generator/snippets/includes.hpp
@@ -12,6 +12,9 @@
 #  include <string.h>  // strnlen
 #  include <string>    // std::string
 #  include <utility>   // std::exchange
+#  if defined( __APPLE__ )
+#    include <cstdlib>  // std::getenv
+#  endif
 #  if 17 <= VULKAN_HPP_CPP_VERSION
 #    include <string_view>
 #  endif

--- a/vulkan/vulkan.hpp
+++ b/vulkan/vulkan.hpp
@@ -18,6 +18,9 @@
 #  include <string.h>  // strnlen
 #  include <string>    // std::string
 #  include <utility>   // std::exchange
+#  if defined( __APPLE__ )
+#    include <cstdlib>  // std::getenv
+#  endif
 #  if 17 <= VULKAN_HPP_CPP_VERSION
 #    include <string_view>
 #  endif

